src/px-script/sidebar/index.js

```diff
@@ -0,0 +1,122 @@
+            const menu = document.createElement("a");
+
+            menu.setAttribute("href", "#");
+            this.hideItems();
+            menu.classList.add("sidebar-openbtn");
+            menu.innerHTML += "<i class='material-icons'>menu</i>";
+            this._el.appendChild(menu);
+
+            menu.addEventListener("click", e => {
+                e.preventDefault();
+
+                if (this.sidebarOpen) {
+                    this.closeSidebar();
+                    this.hideItems();
+                } else {
+                    this.openSidebar();
+                    this.showItems();
+                }
+            });
+        }
+
+        if (this.submenus) {
+            this.submenus.forEach(submenu => {
+                const submenuCopy = submenu.querySelector("i").cloneNode(true);
+
+                submenuCopy.classList.add("submenu-icon-clickable");
+                submenu.insertBefore(submenuCopy, submenu.querySelector("i"));
+                submenuCopy.addEventListener("click", () => {
+                    this.submenuClosed = submenu.classList.contains("submenu-open");
+
+                    if (!this.submenuClosed) {
+                        this.submenuCloseAll();
+                        this.resizeEventSubmenuOpen = this.onResize.bind(this);
+                        window.addEventListener("resize", this.resizeEventSubmenuOpen, { passive: true });
+                        submenu.classList.add("submenu-open");
+                    } else {
+                        this.submenuCloseAll();
+                    }
+                });
+            });
+        }
+    }
+
+    submenuCloseAll () {
+        window.removeEventListener("resize", this.resizeEventSubmenuOpen, { passive: true });
+        this.submenus.forEach(submenu => submenu.classList.remove("submenu-open"));
+    }
+
+    showItems () {
+        [...this._el.querySelectorAll("li")].forEach(listItem => listItem.classList.remove("responsive-hidden"));
+    }
+
+    hideItems () {
+        const firstFour = [...this._el.querySelectorAll("li")].filter(notHidden => !notHidden.classList.contains("responsive-hidden") && notHidden.querySelector(".submenu") === null && this._el.querySelector("UL") === notHidden.parentElement);
+
+        if (this.submenus.length > 0) {
+            this.submenus.forEach(levelTwo => {
+                levelTwo.parentElement.classList.add("responsive-hidden");
+            });
+
+            if (firstFour.length > 4) {
+                firstFour.slice(4).forEach(item => {
+                    item.parentElement.classList.add("responsive-hidden");
+                });
+            }
+        } else {
+            [...this._el.querySelectorAll("li")].slice(4).forEach(items => {
+                items.classList.add("responsive-hidden");
+            });
+        }
+    }
+
+    onResize () {
+        this.closeSidebar();
+        this.hideItems();
+        this.submenuCloseAll();
+    }
+
+    openSidebar () {
```

Comment:
I would rename this method to just `open`, it is a method only used by the `Sidebar` class.
```javascript
const sidebar = new Sidebar();
sidebar.open();
```
Looks better than
```javascript
const sidebar = new Sidebar();
sidebar.openSidebar();
```